Lib/test/test_typing.py

```diff
@@ -10731,6 +10732,107 @@ def test_eq(self):
         with self.assertWarns(DeprecationWarning):
             self.assertNotEqual(int, typing._UnionGenericAlias)
 
+# Define MyType
+class MyType:
+    pass
+
+class TestGenericAliasHandling(BaseTestCase):
+
+    def test_forward_ref(self):
+        fwd_ref = ForwardRef('MyType')
+
+        def func(arg: fwd_ref):
+            pass
+
+        result = get_type_hints(func)
+        self.assertEqual(result['arg'], MyType, f"Expected MyType, got {result['arg']}")
+
+    def test_generic_alias(self):
+        fwd_ref = ForwardRef('MyType')
+        generic_list = List[fwd_ref]
+
+        def func(arg: generic_list):
+            pass
+
+        result = get_type_hints(func)
+        self.assertEqual(result['arg'], List[MyType], f"Expected List[MyType], got {result['arg']}")
```

Comment:
You don't need the "Expected ..." part, the point of assertEqual() is that it makes a nice error message for you already.